From 447c09b2257d839196fb35d8efb4621753d4788f Mon Sep 17 00:00:00 2001
From: Dan Nica <dan.nica@nxp.com>
Date: Mon, 17 Jan 2022 16:14:56 +0200
Subject: [PATCH 52/64] s32gen1: Remove 'SDHC_REDUCED_MAP'

Actually, the latest Zebu models include all uSDHC registers, therefore
the SDHC_REDUCED_MAP macro is no longer needed. But still, some registers
are declared in the driver structure, but not present on s32gen1 platforms,
therefore we'll guard them.

Issue: ALB-4142
Upstream-Status: Pending 

Signed-off-by: Dan Nica <dan.nica@nxp.com>
Signed-off-by: Zhantao Tang <zhantao.tang@windriver.com>
---
 drivers/mmc/fsl_esdhc_imx.c | 12 ++----------
 include/configs/s32-gen1.h  |  4 ----
 2 files changed, 2 insertions(+), 14 deletions(-)

diff --git a/drivers/mmc/fsl_esdhc_imx.c b/drivers/mmc/fsl_esdhc_imx.c
index f1ce633b09..8ce144de72 100644
--- a/drivers/mmc/fsl_esdhc_imx.c
+++ b/drivers/mmc/fsl_esdhc_imx.c
@@ -80,8 +80,8 @@ struct fsl_esdhc {
 	uint    vendorspec;
 	uint    mmcboot;
 	uint    vendorspec2;
-#ifndef SDHC_REDUCED_MAP
 	uint    tuning_ctrl;	/* on i.MX6/7/8/RT */
+#ifndef CONFIG_S32_GEN1
 	char	reserved5[44];
 	uint    hostver;	/* Host controller version register */
 	char    reserved6[4];	/* reserved */
@@ -148,12 +148,10 @@ struct fsl_esdhc_priv {
 	int vs18_enable;
 	u32 flags;
 	u32 caps;
-#if !defined(SDHC_REDUCED_MAP)
 	u32 tuning_step;
 	u32 tuning_start_tap;
 	u32 strobe_dll_delay_target;
 	u32 signal_voltage;
-#endif
 #if CONFIG_IS_ENABLED(DM_REGULATOR)
 	struct udevice *vqmmc_dev;
 	struct udevice *vmmc_dev;
@@ -1138,7 +1136,7 @@ static int esdhc_init_common(struct fsl_esdhc_priv *priv, struct mmc *mmc)
 	esdhc_write32(&regs->dllctrl, 0x0);
 #endif
 
-#if !defined(ARCH_MXC) && !defined(SDHC_REDUCED_MAP)
+#if !defined(ARCH_MXC) && !defined(CONFIG_S32_GEN1)
 	/* Enable cache snooping */
 	esdhc_write32(&regs->scr, 0x00000040);
 #endif
@@ -1368,7 +1366,6 @@ static int fsl_esdhc_init(struct fsl_esdhc_priv *priv,
 	cfg->b_max = CONFIG_SYS_MMC_MAX_BLK_COUNT;
 
 	writel(0, &regs->dllctrl);
-#ifndef SDHC_REDUCED_MAP
 	if (priv->flags & ESDHC_FLAG_USDHC) {
 		if (priv->flags & ESDHC_FLAG_STD_TUNING) {
 			u32 val = readl(&regs->tuning_ctrl);
@@ -1382,7 +1379,6 @@ static int fsl_esdhc_init(struct fsl_esdhc_priv *priv,
 		}
 	}
 
-#endif
 	return 0;
 }
 
@@ -1493,10 +1489,8 @@ static int fsl_esdhc_probe(struct udevice *dev)
 	struct mmc_uclass_priv *upriv = dev_get_uclass_priv(dev);
 	struct fsl_esdhc_plat *plat = dev_get_platdata(dev);
 	struct fsl_esdhc_priv *priv = dev_get_priv(dev);
-#ifndef SDHC_REDUCED_MAP
 	const void *fdt = gd->fdt_blob;
 	int node = dev_of_offset(dev);
-#endif
 	struct esdhc_soc_data *data =
 		(struct esdhc_soc_data *)dev_get_driver_data(dev);
 #if CONFIG_IS_ENABLED(DM_REGULATOR)
@@ -1533,7 +1527,6 @@ static int fsl_esdhc_probe(struct udevice *dev)
 					 ESDHC_FLAG_HS400 |
 					 ESDHC_FLAG_HS400_ES);
 
-#ifndef SDHC_REDUCED_MAP
 	val = fdtdec_get_int(fdt, node, "fsl,tuning-step", 1);
 	priv->tuning_step = val;
 	val = fdtdec_get_int(fdt, node, "fsl,tuning-start-tap",
@@ -1542,7 +1535,6 @@ static int fsl_esdhc_probe(struct udevice *dev)
 	val = fdtdec_get_int(fdt, node, "fsl,strobe-dll-delay-target",
 			     ESDHC_STROBE_DLL_CTRL_SLV_DLY_TARGET_DEFAULT);
 	priv->strobe_dll_delay_target = val;
-#endif
 
 	if (dev_read_bool(dev, "broken-cd"))
 		priv->broken_cd = 1;
diff --git a/include/configs/s32-gen1.h b/include/configs/s32-gen1.h
index 9b6ee072eb..2f6c13460e 100644
--- a/include/configs/s32-gen1.h
+++ b/include/configs/s32-gen1.h
@@ -123,8 +123,4 @@
 #define FSL_QSPI_FLASH_SIZE            SZ_64M
 #endif
 
-#if defined(CONFIG_TARGET_TYPE_S32GEN1_EMULATOR)
-#define SDHC_REDUCED_MAP
-#endif
-
 #endif
-- 
2.17.1

